scripts/start/readCheckScenarioConfig.R

```diff
@@ -0,0 +1,84 @@
+#' read a REMIND scenario_config*.csv file, make sure it contains all columns specified
+#' in path_gdx_list.
+#' Checks whether scenario titles are neither too long, don't contain dots and don't end with a _
+#' Checks for outdated and unknown column titles
+#'
+#' @param filename string with scenario_config*.csv filename
+#' @param remindPath path to remind main directory
+#' @param testmode if TRUE, generates warnings if unknownColumnNames exist
+#' @author Oliver Richters
+#' @return list with scenario config content
+readCheckScenarioConfig <- function(filename, remindPath = ".", testmode = FALSE) {
+  if (testmode) {
+    cfg <- suppressWarnings(gms::readDefaultConfig(remindPath))
+  } else {
+    cfg <- gms::readDefaultConfig(remindPath)
+  }
+  scenConf <- read.csv2(filename, stringsAsFactors = FALSE, row.names = 1, na.strings = "", comment.char = "#")
+  toolong <- nchar(rownames(scenConf)) > 75
+  if (any(toolong)) {
+    warning("These titles are too long: ",
+            paste0(rownames(scenConf)[toolong], collapse = ", "),
+            " – GAMS would not tolerate this, and quit working at a point where you least expect it. Stopping now.")
+  }
+  regionname <- grepl("^(([A-Z]{3})|(glob))$", rownames(scenConf))
+  if (any(regionname)) {
+    warning("These titles may be confused with regions: ",
+            paste0(rownames(scenConf)[regionname], collapse = ", "),
+            " – Titles with three capital letters or 'glob' may be confused with region names by magclass. Stopping now.")
+  }
+  containsdot <- grep("\\.", rownames(scenConf))
+  if (any(containsdot)) {
+    warning("These titles contain a dot: ",
+            paste0(rownames(scenConf)[containsdot], collapse = ", "),
+            " – GAMS would not tolerate this, and quit working at a point where you least expect it. Stopping now.")
+  }
+  underscore <- grep("_$", rownames(scenConf))
+  if (any(underscore)) {
+    warning("These titles end with _: ",
+            paste0(rownames(scenConf)[underscore], collapse = ", "),
+            ". This may lead to wrong gdx files being selected. Stopping now.")
```

Comment:
Many csv used `____whatever____` and `start=0` as "comment". Should we maybe suggest in line 40 that they please properly outcomment using `#`?